cryptofeed/exchanges/huobi.py

```diff
@@ -155,36 +180,196 @@ async def _candles(self, msg: dict, symbol: str, interval: str, timestamp: float
+                "orderPrice":"15000",
+                "orderSize":"0.01",
+                "orderStatus":"filled",
+                "symbol":"btcusdt",
+                "eventType":"trade"
+            }
+        }
+        """
+
+        order = msg['data']
+        if not order['eventType'] in ['creation', 'cancellation', 'trade']:
+            LOG.debug('wrong eventType (does not handle trigger orders yet): %s', msg)
+
+        status = order['orderStatus']
+        if status == 'new':
+            status = SUBMITTING
+        elif status == 'submitted':
+            status = OPEN
+        elif status == 'closed':
+            status = CLOSED
+
+        ts = order.get('orderCreateTime') or order.get('tradeTime') or order.get('lastActTime')
+        oi = OrderInfo(
+            self.id,
+            self.exchange_symbol_to_std_symbol(order['symbol']),
+            str(order['orderId']),
+            BUY if 'buy' in order['type'] else SELL,
+            status,
+            LIMIT if 'limit' in order['type'] else MARKET,
+            Decimal(order['orderPrice']) if 'orderPrice' in order else None,
+            Decimal(order['orderSize']) if 'orderSize' in order else None,
+            Decimal(order['remainAmt']) if 'remainAmt' in order else None,
+            self.timestamp_normalize(ts) if ts else None,
+            raw=msg)
+        await self.callback(ORDER_INFO, oi, timestamp)
+
+    def _auth(self, conn: AsyncConnection):
+        method, host, path = 'GET', conn.conn.host, conn.conn.path
+        ts = datetime.utcnow().strftime('%Y-%m-%dT%H:%M:%S')
+
+        params = {'accessKey': self.key_id, 'signatureMethod': 'HmacSHA256', 'signatureVersion': '2.1', 'timestamp': ts}
+
+        payload = '\n'.join([method, host, path, urlencode(params)])
+        digest = hmac.new(self.key_secret.encode(), payload.encode(), hashlib.sha256).digest()
+
+        signature = base64.b64encode(digest).decode()
+        msg = {
+            'action': 'req',
+            'ch': 'auth',
+            'params': {
+                'authType': 'api',
+                **params,
+                'signature': signature,
+            }
+        }
+
+        return msg
+
+    async def authenticate(self, conn: AsyncConnection):
+        if self.requires_authentication and any(self.is_authenticated_channel(self.exchange_channel_to_std(chan)) for chan in conn.subscription):
+            auth = self._auth(conn)
+            LOG.debug(f"{conn.uuid}: Authenticating with message: {auth}")
+            await conn.write(json.dumps(auth))
+
+    async def message_handler(self, msg: str, conn: AsyncConnection, timestamp: float):
+        if not any(self.is_authenticated_channel(self.exchange_channel_to_std(chan)) for chan in conn.subscription):
+            # unzip message if its from a compressed channel
+            msg = zlib.decompress(msg, 16 + zlib.MAX_WBITS)
         msg = json.loads(msg, parse_float=Decimal)
 
+        LOG.debug(f"{conn.uuid}: Received message: {msg}")
```

Comment:
even in a debug scenario I dont think we want to log every message, that would be a massive amount of volume.